src/api/controllers/misc.ts

```diff
@@ -27,11 +27,31 @@ export async function handleInternalReady(
   const payload = data.d as ReadyPayload;
   setBotID(payload.user.id);
 
-  // Triggered on each shard
+  cache.guildsPartial += payload.guilds.length;
   eventHandlers.shardReady?.(shardID);
+
```

Comment:
All of this, is massively overcomplicating something as simple as an if() with a promise.race to check if anything is missing. IMO even that is overcomplicated a simple 5s delay but this would much morelikely pass as a simple if and not more vars and data eating up memory like this.

Reply:
```ts
const arrived = await Promise.something(payload.guilds.map(g => cacheHandlers.get('guilds', g.id)))
if (!arrived) await delay(5000)
```

Reply:
Something like that